pkg/apply/basic_printer.go

```diff
@@ -23,23 +23,55 @@ type BasicPrinter struct {
 	IOStreams genericclioptions.IOStreams
 }
 
+type applyStats struct {
+	serversideApplied int
+	created           int
+	unchanged         int
+	configured        int
+}
+
+func (a *applyStats) inc(op event.ApplyEventOperation) {
+	switch op {
+	case event.ServersideApplied:
+		a.serversideApplied++
+	case event.Created:
+		a.created++
+	case event.Unchanged:
+		a.unchanged++
+	case event.Configured:
+		a.configured++
+	default:
+		panic(fmt.Errorf("unknown apply operation %s", op.String()))
+	}
+}
+
+func (a *applyStats) sum() int {
+	return a.serversideApplied + a.configured + a.unchanged + a.created
+}
+
 // Print outputs the events from the provided channel in a simple
 // format on StdOut. As we support other printer implementations
 // this should probably be an interface.
 // This function will block until the channel is closed.
 func (b *BasicPrinter) Print(ch <-chan event.Event) {
+	applyStats := &applyStats{}
+	pruneCount := 0
+	deleteCount := 0
 	for e := range ch {
 		switch e.Type {
 		case event.ErrorType:
 			cmdutil.CheckErr(e.ErrorEvent.Err)
 		case event.ApplyType:
 			ae := e.ApplyEvent
 			if ae.Type == event.ApplyEventCompleted {
-				fmt.Fprintf(b.IOStreams.Out, "all resources have been applied\n")
+				fmt.Fprintf(b.IOStreams.Out,
+					"%d resource(s) applied. %d created, %d unchanged, %d configured, %d serverside applied\n",
+					applyStats.sum(), applyStats.created, applyStats.unchanged, applyStats.configured, applyStats.serversideApplied)
```

Comment:
I think end-users may not know about SSA at all so this might leave them wondering.

Reply:
Good point. I have updated the PR so we only print information about ssa if at least one of the resources were applied with ssa.